packages/router/src/lib/request-context.ts

```diff
@@ -0,0 +1,88 @@
+import { TransferState, inject, makeStateKey } from '@angular/core';
+import { HttpHandlerFn, HttpRequest, HttpResponse } from '@angular/common/http';
+
+import { from, of } from 'rxjs';
+
+import { injectBaseURL } from '@analogjs/router/tokens';
+
+/**
+ * Interceptor that is server-aware when making HttpClient requests.
+ * Server-side requests use the full URL
+ * Prerendering uses the internal Nitro $fetch function, along with state transfer
+ * Client-side requests use the window.location.origin
+ *
+ * @param req HttpRequest<unknown>
+ * @param next HttpHandlerFn
+ * @returns
+ */
+export function requestContextInterceptor(
+  req: HttpRequest<unknown>,
+  next: HttpHandlerFn
+) {
```

Comment:
Qq, is this interceptor only for fetching data before loading a page, right?

In the analog app, I have implemented some API routes, and I made a request from an angular service using the HttpClient, and I didn't have to use an interceptor to prepend the full URL; the behavior I assumed was that as long as the angular app has the correct base URL, I don't have to worry about how the URLs are handled in ssr and csr, all work at the same.

I haven't commented before because I haven't tried the SSR data fetching so far https://analogjs.org/docs/features/data-fetching/server-side-data-fetching

Reply:
It's for any request with HttpClient that begins with a `/` meaning its meant to be an "internal" request. 

By default, even if the base href is set in the `index.html`, it won't affect HttpClient requests.

On the server, the full URL is used
One the client, we can use the full URL or just leave it as is
During pre-rendering, API requests remain as-is. Any other requests. Use the full URL.

I'm thinking of including support for passing some header to skip this interceptor also.